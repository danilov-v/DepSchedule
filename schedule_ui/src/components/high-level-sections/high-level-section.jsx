import React from "react";
import { Col } from "reactstrap";

import "./high-level-sections.scss";

export function HighLevelSection({ name, length }) {
  return (
    <Col xs="auto" style={{ width: 40 * length }} className="section">
      {name}
    </Col>
  );
}
