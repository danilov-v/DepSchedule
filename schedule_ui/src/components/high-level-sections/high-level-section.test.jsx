import React from "react";
import { shallow } from "enzyme";
import { HighLevelSection } from "./high-level-section";

describe("high-level-section.jsx", () => {
  describe("<HighLevelSection />", () => {
    test("should render and match snapshot", () => {
      const component = shallow(<HighLevelSection name={"B"} length={4} />);

      expect(component).toMatchSnapshot();
    });

    test("should render and match snapshot with isEmpty", () => {
      const component = shallow(<HighLevelSection isEmpty />);

      expect(component).toMatchSnapshot();
    });
  });
});
