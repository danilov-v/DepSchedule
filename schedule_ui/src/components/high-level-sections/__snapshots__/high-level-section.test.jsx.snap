// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`high-level-section.jsx <HighLevelSection /> should render and match snapshot 1`] = `
<Col
  className="section d-flex align-items-center justify-content-center"
  style={
    Object {
      "width": 160,
    }
  }
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
  xs="auto"
>
  B
</Col>
`;

exports[`high-level-section.jsx <HighLevelSection /> should render and match snapshot with isEmpty 1`] = `
<Col
  className="section d-flex align-items-center justify-content-center empty"
  style={
    Object {
      "width": 40,
    }
  }
  tag="div"
  widths={
    Array [
      "xs",
      "sm",
      "md",
      "lg",
      "xl",
    ]
  }
  xs="auto"
/>
`;
